Sources/Shared/UI/Documents/DocumentsViewController.m

```diff
@@ -995,11 +998,17 @@ -(void)askToMakeFolderActions:(BOOL)createNewFolder{
 
 - (void)sendImageInBackgroundForDirectory:(NSString *)directory data:(NSData *)imageData
 {
-    [_filesProxy fileAction:kFileProtocolForUpload source:directory destination:nil data:imageData];
+    _filesProxy.delegate = self;
+    [_filesProxy uploadFile:imageData asFileName:[directory lastPathComponent] inFolder:fileToApplyAction.currentFolder ofDrive:fileToApplyAction.driveName];
     //Need to reload the content of the folder
+
     [self startRetrieveDirectoryContent];
 }
-
+-(void) fileProxy:(FilesProxy *)fileProxy didUploadImage:(BOOL)finished {
+    if (finished) {
+        [self startRetrieveDirectoryContent];
+    }
```

Comment:
`else {` and show an error message to the user `Upload failed. Please try again later.`